pkg/operators/klusterlet/controllers/ssarcontroller/klusterlet_ssar_controller.go

```diff
@@ -122,36 +129,60 @@ func (c *ssarController) sync(ctx context.Context, controllerContext factory.Syn
 			klusterletNS = klusterletNamespace
 		}
 
-		bootstrapDegradedCondition := checkAgentDegradedCondition(
+		hubConfigDegradedCondition := checkAgentDegradedCondition(
 			ctx, c.kubeClient,
-			bootstrapSecret, bootstrapSecretDegraded,
+			hubConfigSecret, hubConfigSecretDegraded,
 			klusterletAgent{
 				clusterName: klusterlet.Spec.ClusterName,
 				namespace:   klusterletNS,
 			},
-			[]degradedCheckFunc{checkBootstrapSecret},
+			[]degradedCheckFunc{checkHubConfigSecret},
 		)
 
-		hubConfigDegradedCondition := checkAgentDegradedCondition(
+		// the hub kubeconfig is functional, the bootstrap kubeconfig check is not needed,
+		// ignore it to reduce to send the sar requests
+		if hubConfigDegradedCondition.Status == metav1.ConditionFalse {
+			_, _, err := helpers.UpdateKlusterletStatus(
+				ctx,
+				c.klusterletClient,
+				klusterletName,
+				helpers.UpdateKlusterletConditionFn(metav1.Condition{
+					Type:    hubConnectionDegraded,
+					Status:  metav1.ConditionFalse,
+					Reason:  fmt.Sprintf("HubConnectionFunctional"),
+					Message: fmt.Sprintf("Hub connection is functioning correctly"),
+				}),
+			)
+
+			return err
+		}
+
+		bootstrapDegradedCondition := checkAgentDegradedCondition(
 			ctx, c.kubeClient,
-			hubConfigSecret, hubConfigSecretDegraded,
+			bootstrapSecret, bootstrapSecretDegraded,
 			klusterletAgent{
 				clusterName: klusterlet.Spec.ClusterName,
 				namespace:   klusterletNS,
 			},
-			[]degradedCheckFunc{checkHubConfigSecret},
+			[]degradedCheckFunc{checkBootstrapSecret},
 		)
 
-		_, _, err = helpers.UpdateKlusterletStatus(ctx, c.klusterletClient, klusterletName,
-			helpers.UpdateKlusterletConditionFn(bootstrapDegradedCondition),
-			helpers.UpdateKlusterletConditionFn(hubConfigDegradedCondition))
-
-		if err != nil {
-			klog.Errorf("Update Klusterlet Status Failed: %v", err)
-		}
-	}()
+		// merge the bootstrap and hubconfig degraded conditions to hub connection degraged condition
+		_, _, err := helpers.UpdateKlusterletStatus(
+			ctx,
+			c.klusterletClient,
+			klusterletName,
+			helpers.UpdateKlusterletConditionFn(metav1.Condition{
+				Type:    hubConnectionDegraded,
+				Status:  metav1.ConditionTrue,
+				Reason:  bootstrapDegradedCondition.Reason + "," + hubConfigDegradedCondition.Reason,
+				Message: bootstrapDegradedCondition.Message + "\n" + hubConfigDegradedCondition.Message,
+			}),
+		)
+		return err
+	})
 
-	return nil
+	return errs.Wait()
```

Comment:
this makes it locked until error returns? I think we want to ensure that when a klusterlet comes, it spawns a goroutine and not wait anymore. So the check in this gorountine can be pretty slow due to the set on qps.

Reply:
yes, this will wait the error return, but I think this will not affect the creation of  goroutine, because when a klusterlet comes, there will be an event, this event will trigger a new call to the sync method, am I right?

Reply:
changed to when error is happened, we requeue the request